include/devicetree.h

```diff
@@ -1,15 +1,1283 @@
 /*
  * SPDX-License-Identifier: Apache-2.0
  * Copyright (c) 2020 Nordic Semiconductor
+ * Copyright (c) 2020, Linaro Ltd.
  *
  * Not a generated file. Feel free to modify.
  */
 
+/**
+ * @file
+ * @brief Devicetree main header
+ *
+ * API for accessing the current application's devicetree macros.
+ */
+
 #ifndef DEVICETREE_H
 #define DEVICETREE_H
 
 #include <devicetree_unfixed.h>
 #include <devicetree_legacy_unfixed.h>
 #include <devicetree_fixups.h>
 
+#include <sys/util.h>
+
+/**
+ * @brief devicetree.h API
+ */
+
+/*
+ * Property suffixes
+ * -----------------
+ *
+ * These are the optional parts that come after the _P_<property>
+ * part in DT_N_<path-id>_P_<property-id> macros, or the "prop-suf"
+ * nonterminal in the DT guide's macros.bnf file.
+ *
+ * Before adding new ones, check this list to avoid conflicts. If any
+ * are missing from this list, please add them. It should be complete.
+ *
+ * _EXISTS: property is defined
+ * _LEN: property logical length
+ * _IDX_<i>: logical index into property
+ * _IDX_<i>_VAL_<val>: phandle array's specifier value by index
+ * _IDX_<i>_VAL_<val>_EXISTS: cell value exists, by index
+ * _IDX_<i>_PH: phandle array's phandle by index
+ * _NAME_<name>_VAL_<val>: phandle array's property specifier by name
+ * _NAME_<name>_VAL_<val>_EXISTS: cell value exists, by name
+ * _NAME_<name>_PH: phandle array's phandle by name
+ * _ENUM_IDX: property's value as an index into bindings enum
```

Comment:
It would be easier to search for conflicts (and interpretation) if these were in alphabetical order.